plugins/ltac/extratactics.mlg

```diff
@@ -146,6 +146,23 @@ let discrHyp id =
 let injection_main with_evars c =
  elimOnConstrWithHoles (injClause None None) with_evars c
 
+let warn_deprecated_injection_intropattern_syntax = fun ?loc () -> ()
+(* Keeping code in case the general view evolves (e.g. when "on H do pat"
+   is introduced) *)
+(*
+  CWarnings.create ~name:"injection-arguments-flat-intropattern-syntax" ~category:"deprecated"
+    (fun () ->
+       strbrk "Syntax " ++ str "\"injection ... as pat1 ... patn\"" ++
+       strbrk " is deprecated. Use " ++ str "\" injection ... as [= pat1 ... patn ]\".")
+*)
+
+let check_inj_ipat ?loc = function
+  | [{ CAst.v = IntroAction (IntroInjection ipat) }] -> ipat
+  | [] -> warn_deprecated_injection_intropattern_syntax (); []
```

Comment:
Moreover, this is confusing, because if you don't look further, you may think that this is actually deprecated. And it complexifies the code.